### PR TITLE
Fix bugs reported by mercury in batch #2

### PR DESCRIPTION
**Pull request by rubio41**
Bugs fixed:

- [x]  Tuesday in the morning
- [x]   next week on Friday
- [x]   late morning Tuesday
- [x]   midnight
- [x]   which emails have got[ten a] reply, ten a should not be extracted
- [x]   In a week, for an hour
- [x]   over the weekend

### CODE REVIEW

File: Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs [hunk 1]
```diff
@@ -69,26 +72,29 @@ public bool GetMatchedTimeRange(string text, out string timeStr, out int beginHo
             beginHour = 0;
             endHour = 0;
             endMin = 0;
-            if (trimedText.EndsWith("morning"))
+            if (trimedText.EndsWith("morning") || trimedText.StartsWith("morning"))
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs]: Don't repeat strings. Declare a constant and re-use it. Or create a regex if more are likely to be added later.

File: Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs [hunk 2]
```diff
@@ -348,16 +348,51 @@ protected virtual DateTimeResolutionResult ParseSpecificNight(string text, DateO
...
             int beginHour, endHour, endMin = 0;
             string timeStr;
-            if (!this.Config.GetMatchedTimeRange(trimedText, out timeStr, out beginHour, out endHour, out endMin))
+            if (!this.Config.GetMatchedTimeRange(timeText, out timeStr, out beginHour, out endHour, out endMin))
             {
                 return ret;
             }
 
-            var match = this.Config.SpecificNightRegex.Match(trimedText);
+            // modify time period if "early" or "late" is existed
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs]: Can you explain the logic behind this? Just from the code it's hard to figure out what exactly is happening.

File: Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs [hunk 3]
```diff
@@ -376,21 +411,26 @@ protected virtual DateTimeResolutionResult ParseSpecificNight(string text, DateO
                 return ret;
             }
 
-
             // handle Date followed by morning, afternoon
-            match = this.Config.NightRegex.Match(trimedText);
+            // handle morning, afternoon followed by Date
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs]: What about "noon"?

File: Microsoft.Recognizers.Text.DateTime.Tests/English/TestTimeParser.cs [hunk 4]
```diff
@@ -128,6 +143,20 @@ public void TestTimeParseLuis()
...
             BasicTest("I'll be back 11ish", "T11");
             BasicTest("I'll be back 11-ish", "T11");
+
+            BasicTest("midnight", "T00");
+            BasicTest("mid-night", "T00");
+            BasicTest("mid night", "T00");
+            BasicTest("midmorning", "T10");
+            BasicTest("mid-morning", "T10");
+            BasicTest("mid morning", "T10");
+            BasicTest("midafternoon", "T14");
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime.Tests/English/TestTimeParser.cs]: Can we add a comment to the code mentioning exactly where on the spec is the definition of afternoon?
It's very counter intuitive that midafternoon resolves to T14.

File: Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs (hunk 3 above)
Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs]: noon is defined as 12pm. It is supported before.

File: Microsoft.Recognizers.Text.DateTime.Tests/English/TestTimeParser.cs (hunk 4 above)
Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime.Tests/English/TestTimeParser.cs]: add comment and TODO for that

File: Microsoft.Recognizers.Text.DateTime.Tests/English/TestTimeExtractor.cs [hunk 5]
```diff
@@ -16,6 +16,8 @@ public void BasicTest(string text, int start, int length)
             Assert.AreEqual(Constants.SYS_DATETIME_TIME, results[0].Type);
         }
 
+
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime.Tests/English/TestTimeExtractor.cs]: Nitpick: Don't add multiple blank lines.

**Pull request merged**